rmw_fastrtps_shared_cpp/src/rmw_response.cpp

```diff
@@ -104,6 +104,22 @@ __rmw_send_response(
   wparams.related_sample_identity().sequence_number().low =
     (int32_t)(request_header->sequence_number & 0xFFFFFFFF);
 
+  // According to the list of possible entity kinds in section 9.3.1.2 of RTPS
+  // readers will have this bit on, while writers will not. We use this to know
+  // if the related guid is the request writer or the response reader.
+  constexpr uint8_t entity_id_is_reader_bit = 0x04;
+  const eprosima::fastrtps::rtps::GUID_t & related_guid =
+    wparams.related_sample_identity().writer_guid();
+  if((related_guid.entityId.value[3] & entity_id_is_reader_bit) != 0)
+  {
+    // Related guid is a reader, so it is the response subscription guid.
+    // Wait for the response writer to be matched with it.
+    if(!info->pub_listener_->wait_for_subscription(related_guid, std::chrono::milliseconds(100))) {
```

Comment:
`rmw_send_response` was not blocking, and it should be continue being not blocking.

Reply:
What if we change this to `0` timeout? That way we at least know that it won't work and can react to it.

Reply:
Based on the conversation below, I assume the code is ok as it is right now.

If this be changed to an indefinite block, please tell me and I'll do it